weapon-config.inc

```diff
@@ -5213,7 +5236,7 @@ public OnPlayerDamageDone(playerid, Float:amount, issuerid, weapon, bodypart)
 		}
 	}
 
-	if (s_DamageFeed) {
+	if (s_DamageFeed || !s_DamageFeed && s_DamageFeedPlayer[playerid]) {
```

Comment:
This line is partially pointless (as is line 4736).  If `s_DamageFeed` fails then `!s_DamageFeed` MUST pass.  I'm going to add brackets, as they make the precedence explicit (always a good idea - you might want to consider that in the future even if you know that `&&` is higher than `||`, simply as others may not):

```pawn
if ((s_DamageFeed) || (!s_DamageFeed && s_DamageFeedPlayer[playerid]))
```

Now, I can understand WHY you might want both versions in - you want the next code to run EITHER when `s_DamageFeed` is set, or when `s_DamageFeedPlayer` is set, but the latter only when `s_DamageFeed` is NOT set.  Given that, what is written seems sensible, but because of "short-circuiting", it isn't.

Short-circuiting is when only part of an expression is run - enough to determine the answer, but not too much as to waste time:

```pawn
F1()
{
    print("1");
    return 1;
}

F2()
{
    print("2");
    return 2;
}

main()
{
    if (F1() || F2())
    {
        print("3");
    }
}
```

You might expect the output in this case to be:

```
1
2
3
```

With an execution order like this:

* Call `F1()` and get the return value.
* Call `F2()` and get the return value.
* Check if either value was non-zero.
* If so (in this case, they are) enter then "then" statement.

However, the actual output is:

```
1
3
```

With an execution order of:

* Call `F1()` and get the return value.
* Check if it was non-zero.
* If so (in this case, it is) enter then "then" statement.

`F2()` is simply never called.  When using `||`, only one of the conditions needs to be true.  If the first one is true, it doesn't matter if the second one is true, so there's no point in checking.  Now, if you have side-effects in the second function that you expect to see (such as that print), then this may not be what is wanted (in which case you can use `|` instead).

But going back to the original code, if `(s_DamageFeed)` is true, then the second half will never be checked at all.  If the second half IS being checked, it MUST be because `(s_DamageFeed)` failed, and so by that point you KNOW that `!s_DamageFeed` is true - if you know it is true, don't check it again:

```pawn
if ((s_DamageFeed) || (s_DamageFeedPlayer[playerid]))
```

As a side-note, are you aware that your commit dropped some trailing tabs on entirely unrelated blank lines?  I don't know how Slice feels about that - some people get very territorial about spacing, some people couldn't care less.  However, it is often a side-effect of using a bad editor (read: Pawno) and just something to watch for.